documentation/sse_user_manual.md

```diff
@@ -0,0 +1,117 @@
+
+## Usage
+Consider a scenario with 3 different documents identified as `id1`, `id2`, `id3`, respectively. The documents contain, among other words, the following keywords, and the keyword/identifier pairs can be added to the secure index.
+
+* `id1` contains the keyword `keyword1`. 
+* `id1` and `id2` both contain the keyword `keyword2`.
+* `id2` contains the keyword `keyword3`.
+* `id1`, `id2`, and `id3` all contain the keyword `keyword4`.
+
+Given a keyword for a `Search` query, all the identifiers that contain the given keyword can then be identified, even when the documents are encrypted. Below, a secure index is initialized, and the keyword/identifier pairs listed above are added to the secure index.
+
+```go
+secureIndex := NewSecureIndex(&keyProvider, &ioProvider, &idProvider)
+```
+Note that a valid `token` must be provided. A `token` is valid if the caller can use it to be authenticated by the Identity Provider and to be authorized to use the secure index.
+
+```go
+if err := secureIndex.Add(token, "keyword1", "id1"); err != nil {
+    return err
+}
+if err = secureIndex.Add(token, "keyword2", "id1"); err != nil {
+    return err
+}
+if err = secureIndex.Add(token, "keyword2", "id2"); err != nil {
+    return err
+}
+if err = secureIndex.Add(token, "keyword3", "id2"); err != nil {
+    return err
+}
+if err = secureIndex.Add(token, "keyword4", "id1"); err != nil {
+    return err
+}
+if err = secureIndex.Add(token, "keyword4", "id2"); err != nil {
+    return err
+}
+if err = secureIndex.Add(token, "keyword4", "id3"); err != nil {
+    return err
+}
+```
+
+A `Search` query can be used to identify which documents contain e.g. `keyword4`:
+
+```go
+IDs, err := secureIndex.Search(token, "keyword4")
+if err != nil {
+    return err
+}
+
+fmt.Println(IDs)
+```
+
+```go
+Out:    ["id1", "id2", "id3"]
+```
+
+A keyword can be deleted from a document, below `keyword4` is deleted from `id1`.
+```go
+err = secureIndex.Delete(token, "keyword4", "id1")
+if err != nil {
+    return err
+}
+```
+
+After the `Delete` query, the `Search` query will now return a different output:
+```go
+IDs, err = secureIndex.Search(token, "keyword4")
+if err != nil {
+    return err
+}
+
+fmt.Println(IDs)
+```
+
+```go
+Out:    ["id2", "id3"]
+```
+
+## Implementation
+
+In this section, some more technical details about the implementation are given.
+
+Given a keyword and an identifer for an `Add` query, a label is computed based on the keyword as well as on a *counter* (explained below), and an Identifier struct representing the identifier is created. The label is then mapped to the Identifier as shown below, and the "label -> Identifier" correlation is stored in the secure index. The Identifier is sealed before it is stored in order to avoid having plaintext keywords or identifiers outside of the D1 Library. Note that the keyword is used to seal the Identifier which means that the sealed Identifier can only be unsealed if the keyword is known.
```

Comment:
```suggestion
Given a keyword and an identifer for an `Add` query, a label is computed based on a secret key, the keyword, and a *counter* (explained below), and an Identifier struct representing the identifier is created. The label is then mapped to the Identifier as shown below, and the "label -> Identifier" correlation is stored in the secure index. The Identifier is sealed before it is stored in order to avoid having plaintext keywords or identifiers outside of the D1 Library. Note that the keyword is used to seal the Identifier which means that the sealed Identifier can only be unsealed if the keyword is known.
```